Makefile

```diff
@@ -585,8 +587,17 @@ rpm: tarball test
 
 # Removing test dependency until resolved
 tarball:
-	VERSION=`awk '/^Version/ {print $$2}' deepsea.spec`; \
-	git archive --prefix deepsea-$$VERSION/ -o ~/rpmbuild/SOURCES/deepsea-$$VERSION.tar.bz2 HEAD
+	mkdir -p .tmp
+	git archive --prefix deepsea-tmp/ -o .tmp/deepsea-tmp.tar HEAD
+	HASH=`git rev-parse --short HEAD`; \
+	cd .tmp; tar -xf deepsea-tmp.tar; cd deepsea-tmp; \
+	VERSION=`cat version.txt`; \
+	echo "$$VERSION+git.$$HASH" > version.txt; \
```

Comment:
This is problematic because it does not guarantee that each successive build will have an ascending version number. You probably want to add a "git offset" component which can be obtained by parsing the output of `git describe`.

I thought `version.txt` was going to include just the "upstream" version number (i.e. "0.8" - shouldn't this be "0.8.0" though?) and the git offset + sha1 part would be appended to it in the packaging?

Reply:
```
+	VERSION=`cat version.txt`; \
+	echo "$$VERSION+git.$$HASH" > version.txt; \
```

whoa, doesn't this overwrite `version.txt`? What happens if I run "make tarball" 3 times in a row?

Reply:
ah, nevermind, it's doing it in a temporary directory. So, ignore all the comments above, except for the "ascending version number" one :-)

Reply:
Note that RPM will refuse to upgrade version "0.8+git.81234567" to "0.8+git.12345678" even if the latter is newer (git SHA1s are essentially random).

Also, once you are parsing `git describe`, you might as well get the SHA1 from that, too.

Reply:
My problem with the getting the offset is, which base should I compute the offset from?
The last defined tag?

Reply:
Indeed, using `git describe` assumes that the "upstream" version number will be incremented whenever a new tag is added - otherwise the addition of any new tag causes the offset to be reset to zero. Very good point. If you want something foolproof, use "seconds since epoch" instead of the offset.

I was/am just in the process of switching the ceph OBS build from "seconds since epoch" to "offset" but now I see we're going to have the same problem there.

Reply:
I pushed an offset computation that computes the offset since the last commit that changed the `version.txt` file. That way it is not dependent on a git tag but rather on the version bump commit